src/Dictionaries/RedisDictionarySource.cpp

```diff
@@ -250,23 +222,72 @@ namespace DB
                 if (isInteger(type))
                     key << DB::toString(key_columns[i]->get64(row));
                 else if (isString(type))
-                    key << get<String>((*key_columns[i])[row]);
+                    key << get<const String &>((*key_columns[i])[row]);
                 else
                     throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected type of key in Redis dictionary");
             }
 
             keys.add(key);
         }
 
-        return Pipe(std::make_shared<RedisSource>(client, std::move(keys), storage_type, sample_block, max_block_size));
+        return Pipe(std::make_shared<RedisSource>(std::move(conn), std::move(keys), storage_type, sample_block, REDIS_MAX_BLOCK_SIZE));
     }
 
-
     String RedisDictionarySource::toString() const
     {
         return "Redis: " + host + ':' + DB::toString(port);
     }
 
+    auto RedisDictionarySource::getConnection() const -> ConnectionPtr
```

Comment:
Consider to use `ConnectionPtr` as return type, without this auto.

Reply:
This construction avoid extra qualification for `ConnectionPtr`. Otherwise we will have to write ````RedisDictionarySource::ConnectionPtr RedisDictionarySource::getConnection() const```

But I'm ok to replace this.